source/beamline/BioXAS/BioXASBeamline.cpp

```diff
@@ -338,6 +338,19 @@ CLSValves* BioXASBeamline::valves() const
 	return result;
 }
 
+AMXRFDetector *BioXASBeamline::xrfDetector() const
```

Comment:
the overloaded version should be the one you need to use when you want a specific detector. This function might also be potentially generalized.